packages/block-editor/src/utils/dom.js

```diff
@@ -134,41 +134,50 @@ function isScrollable( element ) {
 	);
 }
 
+export const WITH_OVERFLOW_ELEMENT_BLOCKS = [ 'core/navigation' ];
 /**
- * Returns the rect of the element including all visible nested elements.
- *
- * Visible nested elements, including elements that overflow the parent, are
- * taken into account.
- *
- * This function is useful for calculating the visible area of a block that
- * contains nested elements that overflow the block, e.g. the Navigation block,
- * which can contain overflowing Submenu blocks.
+ * Returns the rect of the element.
  *
+ * Note: the function handles special cases for blocks that contain visible,
+ * nested elements that overflow the block, e.g. the Navigation block,
+ * which can have overflowing submenu blocks. Such blocks are defined in
+ * `WITH_OVERFLOW_ELEMENT_BLOCKS`. For such blocks,
+ * the bounds of the visible children are calculated to determine
+ * the full extent of the block including the visible children that overflow the parent.
```

Comment:
```suggestion
 * Returns the bounding rectangle of an element, with special handling for blocks
 * that have visible overflowing children (defined in WITH_OVERFLOW_ELEMENT_BLOCKS).
 *
 * For blocks like Navigation that can have overflowing elements (e.g. submenus),
 * this function calculates the combined bounds of both the parent and its visible
 * children. The returned rect may extend beyond the viewport.
```

Nit: It's probably just the fact that my brain is still warming up on a Monday morning but this comment didn't read cleanly to me. Would something like the above be better? Your call.